http/handler.go

```diff
@@ -464,6 +489,9 @@ func wrapGenericHandler(core *vault.Core, h http.Handler, props *vault.HandlerPr
 
 		h.ServeHTTP(w, r)
 
+		// deleting the in-flight request entry
+		core.DeleteInFlightReqData(inFlightReqID)
```

Comment:
Why not do this in a defer instead?  That way if something should cause the preceding code to return early, we're sure not to leak map entries.